### ISSUE TITLE
[wasmer] panic! triggered in lib/runtime-core/src/types.rs

### ISSUE THREAD

**Issue by pventuzelo** [wasmerio/wasmer#653]
Thanks for the bug report!

### Describe the bug

panic! is called when the memory is shared but without max value given. 
This issue is making wasmer to panicked.

https://github.com/wasmerio/wasmer/blob/f68ad7f655c3d219c9631a347b8379a69c33f939/lib/runtime-core/src/types.rs#L340-L348

### Expected behavior

`wasmer` should generate an exception and quit properly.

### Status of my environment

```bash
echo "`wasmer -V` | `rustc -V` | `uname -m`"
wasmer 0.6.0 | rustc 1.38.0-nightly (0b680cfce 2019-07-09) | x86_64
```

### Steps to reproduce

Download [panic_shared_memory_without_max.zip](https://github.com/wasmerio/wasmer/files/3485777/panic_shared_memory_without_max.zip)

```
$ unzip panic_shared_memory_without_max.zip
$ wasmer run panic_shared_memory_without_max.wasm
thread 'main' panicked at 'shared memory without a max is not allowed', lib/runtime-core/src/types.rs:346:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

### Actual behavior

```
$ RUST_BACKTRACE=1 wasmer run panic_shared_memory_without_max.wasm

thread 'main' panicked at 'shared memory without a max is not allowed', lib/runtime-core/src/types.rs:346:30
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:212
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:475
   7: std::panicking::begin_panic
   8: wasmer_runtime_core::types::MemoryDescriptor::memory_type
   9: <wasmer_clif_backend::code::FunctionEnvironment as wasmer_clif_fork_wasm::environ::spec::FuncEnvironment>::make_heap
  10: wasmer_clif_fork_wasm::state::TranslationState::get_heap
  11: wasmer_clif_fork_wasm::code_translator::translate_load
  12: wasmer_clif_fork_wasm::code_translator::translate_operator
  13: <wasmer_clif_backend::code::CraneliftFunctionCodeGenerator as wasmer_runtime_core::codegen::FunctionCodeGenerator<wasmer_clif_backend::code::CodegenError>>::feed_event
  14: wasmer_runtime_core::codegen::MiddlewareChain::run
  15: wasmer_runtime_core::parse::read_module
  16: <wasmer_runtime_core::codegen::StreamingCompiler<MCG,FCG,RM,E,CGEN> as wasmer_runtime_core::backend::Compiler>::compile
  17: wasmer_runtime_core::compile_with_config
  18: wasmer_runtime::compile_with_config_with
  19: wasmer::execute_wasm
  20: wasmer::main
  21: std::rt::lang_start::{{closure}}
  22: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:49
  23: std::panicking::try::do_call
             at src/libstd/panicking.rs:294
  24: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:85
  25: std::panicking::try
             at src/libstd/panicking.rs:273
  26: std::panic::catch_unwind
             at src/libstd/panic.rs:388
  27: std::rt::lang_start_internal
             at src/libstd/rt.rs:48
  28: main
  29: __libc_start_main
  30: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Additional context

The wasm module attached is detect as invalid by multiple other wasm tools
```
$ wasm-validate panic_shared_memory_without_max.wasm 
0000141: error: shared memory must have a max size
$  wasm-opt panic_shared_memory_without_max.wasm 
[parse exception: shared memory must have max size (at 0:321)]
Fatal: error in parsing input
```

**Comment by pventuzelo** [comment]
TODO: add fuzzer-trophy tag for this issue

**Comment by pventuzelo** [comment]
I tried pull request #654 and it's good for me.

```
$ git checkout upstream/feature/fix-memory-type-panic
$ make release
$ ./target/release/wasmer run reported_issues/panic_shared_memory_without_max.wasm
execute_wasm: "Can\'t compile module: InternalError { msg: \"Codegen(\\\"\\\\\\\"Max number of pages is required for shared memory\\\\\\\"\\\")\" }"
```

**Comment by pventuzelo** [comment]
It's seems that the patch of this issue was not merge to master (or reintroduce).

**Comment by pventuzelo** [comment]
Fixed here: 42e4871266b9fda6554ce310733b6d554483bc2d
